### ISSUE TITLE
In _attachments revpos values are keep changing when do pull replication

### ISSUE THREAD

**Issue by prasannakumark** [couchbase/couchbase-lite-android#994]
Hi All,

I've observed that when i do the pull replications, the documents of the attachments values(revpos) are keep changing in local documents. Let me explain with an example.
Current document in server...

> {
>   "_id": "0B81E1E3-580A-4EE7-8A70-DA0CB7ABB620",
>    "_rev": "51-f15a19985584cd68fc237a367727a36e",
>   "archived":null,
>  "_attachments": {
>        "20141220-075930961.256x192.jpg": {
>            "content_type": "image/jpeg",
>            "revpos": 1,
>            "digest": "md5-+nbZfjy+lrNhyDfmUmvGuw==",
>            "length": 14704,
>            "stub": true
>        },
>        "20141220-075930961.jpg": {
>            "content_type": "image/jpeg",
>            "revpos": 1,
>            "digest": "md5-U9SSER6xHgEdX5MHGP979g==",
>            "length": 126070,
>            "stub": true
>        },
>        "20141220-075926857.png": {
>            "content_type": "image/png",
>            "revpos": 1,
>            "digest": "md5-e4+Sp5jaa1AlhE51TgiWPQ==",
>            "length": 175062,
>            "stub": true
>        },
>        "20141220-075926857.256x192.png": {
>            "content_type": "image/png",
>            "revpos": 1,
>            "digest": "md5-+8OaZao5SKQZj2wQnkwK2w==",
>            "length": 18381,
>            "stub": true
>        }
>    }
> }

Now above document having "revpos": 1(in _attachments JSON object), when i start pull replication in mobile side the revpos is changing to "revpos": 51(I mean to current rev id). so the problem here is when i do any document update in mobile side that document is not pushing to server getting 412 error

As the information is missing for the couchDB, It says as an error (412)
Example Error:  {
    "error": "missing_stub",
    "reason": "id:0B81E1E3-580A-4EE7-8A70-DA0CB7ABB620, name:20141220-075930961.jpg"
}

After understanding the exact problem i took the CBL Library code start the debugging than I understood that **stub** is missing(in attachInfo) in that code, Due to this reason its adding new **generation** value for revpos.

**What I changed to make work?**

 In DataBase class -> processAttachmentsForRevision()
https://friendpaste.com/6n8wQIYv7u5iRTVg1JoX8e
 //This what i added
 attachInfo.put("stub",true);

Why did I add this value is in **AttachmentInternal** contructor we are passing the **attachInfo** object in that there is an condition tell us 

```
else if (attachInfo.containsKey("stub") &&
                ((Boolean) attachInfo.get("stub")).booleanValue()) {
            // This item is just a stub; validate and skip it
            if (attachInfo.containsKey("revpos")) {
                int revPos = ((Number) attachInfo.get("revpos")).intValue();
                if (revPos <= 0) {
                    throw new CouchbaseLiteException(Status.BAD_ATTACHMENT);
                }
                setRevpos(revPos);
            }
            // skip
        }
```

Means is stub key is there than set the old revpos only(setRevpos(revPos);) or else set new **generation** value.

Now I would say every think working perfectly.

Thanks 
Prasanna Kumar

**Comment by rinverardi** [comment]
We are seeing this on Android as well. It seems the same bug was corrected on iOS but the fix never made it into the Java code:

https://github.com/couchbase/couchbase-lite-ios/commit/ef0dbd53e56c0dc8e7bf354abd1f07f215c0c864

@djpongh, @hideki This has been lingering around for well over a year. If we provided a fix and a corresponding unit test, what are the chances of it being included in the next Couchbase Lite version, and at what time can we expect the next version?
